src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -384,11 +419,22 @@ class RemoteMachineTrainingService implements TrainingService {
         // Copy NNI scripts to remote expeirment working directory
         const remoteScriptsDir: string = this.getRemoteScriptsPath();
         await SSHClientUtility.remoteExeCommand(`mkdir -p ${remoteScriptsDir}`, conn);
-        await SSHClientUtility.copyDirectoryToRemote('./scripts', remoteScriptsDir, conn, this.remoteOS);
         await SSHClientUtility.remoteExeCommand(`chmod 777 ${nniRootDir} ${nniRootDir}/* ${nniRootDir}/scripts/*`, conn);
-
+        
+        const gpuCollectorContent: string = String.Format(
```

Comment:
Suggest to create only one copy of the gpu_metrics_collector.sh on local if the multiple copies are identical.

Reply:
remove generate gpu shell to generateGpuScript() and only create once.